lib/open-uri.rb

```diff
@@ -234,6 +236,7 @@ def OpenURI.open_loop(uri, options) # :nodoc:
         uri = redirect
         raise "HTTP redirection loop: #{uri}" if uri_set.include? uri.to_s
         uri_set[uri.to_s] = true
+        raise "Too many redirects" if max_redirects && uri_set.size > max_redirects
```

Comment:
We probably should have a specific exception class for this, so users that want to catch this don't have to rescue RuntimeError.